### PR TITLE
feat(core): add track events for editor header

### PR DESCRIPTION
**Pull request by JimmFly**
close AF-1054

### CODE REVIEW

**Comment by JimmFly** [comment]
* **#7668** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7668?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7667** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7667?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7664** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7664?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7661** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7661?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#7735** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7735?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7695** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7695?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @JimmFly and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by CatsJuice — COMMENTED** [review]
Should track https://github.com/toeverything/AFFiNE/pull/7652?

**Comment by JimmFly** [comment]
> Should track #7652?

I will rebase

File: packages/frontend/core/src/pages/workspace/detail-page/detail-page-header.tsx [hunk 1]
```diff
@@ -115,6 +125,16 @@ export function NormalPageHeader({ page, workspace }: PageHeaderProps) {
   const onRename = useCallback(() => {
     setTimeout(() => titleInputHandleRef.current?.triggerEdit());
   }, []);
+
+  const onTriggerEdit = useCallback(() => {
+    mixpanel.track('header option click', {
```
Comment (pengx17) [inline packages/frontend/core/src/pages/workspace/detail-page/detail-page-header.tsx]: the tracked event name should be camel-case, like  `HeaderOptionClick`

Comment (JimmFly) [reply packages/frontend/core/src/pages/workspace/detail-page/detail-page-header.tsx]: I also changed the event name in the relevant PR.

File: packages/frontend/core/src/components/blocksuite/block-suite-mode-switch/index.tsx [hunk 2]
```diff
@@ -104,6 +104,15 @@ export const EditorModeSwitch = ({
     [currentMode, isPublic]
   );
 
+  const onModeSwitch = useCallback(() => {
+    mixpanel.track('HeaderOptionClick', {
```
Comment (EYHN) [inline packages/frontend/core/src/components/blocksuite/block-suite-mode-switch/index.tsx]: put this in `onSwitchToPageMode` and `onSwitchToEdgelessMode`

**Review by pengx17 — COMMENTED** [review]
restack on to https://github.com/toeverything/AFFiNE/pull/7681

File: packages/frontend/core/src/components/blocksuite/block-suite-mode-switch/index.tsx [hunk 3]
```diff
@@ -71,27 +71,38 @@ export const EditorModeSwitch = ({
...
+      module: 'editor header',
+      control: 'switch mode',
+    });
+  }, [currentMode]);
+
   const onSwitchToPageMode = useCallback(() => {
     mixpanel.track('Button', {
       resolve: 'SwitchToPageMode',
     });
+    onModeSwitch();
```
Comment (forehalo) [inline packages/frontend/core/src/components/blocksuite/block-suite-mode-switch/index.tsx]: why two events in switch mode

File: packages/frontend/core/src/mixpanel/events.ts [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+    paywall: {
+      storage: ['viewPlans'];
+      aiAction: ['viewPlans'];
+    };
+    header: {
+      actions: [
+        'createDoc',
+        'createWorkspace',
+        'switchPageMode',
+        'docOptionsEntry',
```
Comment (forehalo) [inline packages/frontend/core/src/mixpanel/events.ts]: verb

File: packages/frontend/core/src/mixpanel/events.ts [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+      ];
+      share: ['createShareLink', 'copyShareLink', 'export'];
+      docOptions: [
+        'viewHistoryVersions',
+        'viewInfo',
+        'moveToTrash',
+        'rename',
+        'switchPageMode',
+        'duplicate',
```
Comment (forehalo) [inline packages/frontend/core/src/mixpanel/events.ts]: duplicate => createDoc({ control: 'duplicate' })

File: packages/frontend/core/src/components/page-list/operation-menu-items/export.tsx [hunk 6]
```diff
@@ -115,6 +116,16 @@ export const Export = ({ exportHandler, className, pageMode }: ExportProps) => {
       pageMode={pageMode}
     />
   );
+  const handleExportMenuOpenChange = useCallback((open: boolean) => {
+    if (open) {
+      mixpanel.track('PageOptionClick', {
```
Comment (forehalo) [inline packages/frontend/core/src/components/page-list/operation-menu-items/export.tsx]: update

File: packages/frontend/core/src/components/blocksuite/block-suite-header/favorite/index.tsx [hunk 7]
```diff
@@ -30,11 +31,16 @@ export const useFavorite = (pageId: string) => {
 export const FavoriteButton = ({ pageId }: FavoriteButtonProps) => {
   const { favorite, toggleFavorite } = useFavorite(pageId);
 
+  const handleFavorite = useCallback(() => {
+    track.$.header.actions.toggleFavorite();
```
Comment (forehalo) [inline packages/frontend/core/src/components/blocksuite/block-suite-header/favorite/index.tsx:35]: simple event can tracked by data-event-props

File: packages/frontend/core/src/mixpanel/events.ts [hunk 8]
```diff
@@ -35,7 +35,13 @@ type DocEvents =
   | 'renameDoc'
   | 'linkDoc'
   | 'deleteDoc'
-  | 'switchPageMode';
+  | 'switchPageMode'
+  | 'openDocOptionsMenu'
```
Comment (forehalo) [inline packages/frontend/core/src/mixpanel/events.ts:39]: use `open` event and put in `share` module

File: packages/frontend/core/src/mixpanel/events.ts [hunk 9]
```diff
@@ -35,7 +35,13 @@ type DocEvents =
...
+  | 'openDocInfo'
+  | 'editDocTitle'
+  | 'moveToTrash'
```
Comment (forehalo) [inline packages/frontend/core/src/mixpanel/events.ts]: deleteDoc

File: packages/frontend/core/src/pages/workspace/detail-page/detail-page-header.tsx [hunk 10]
```diff
@@ -142,6 +146,7 @@ export function NormalPageHeader({ page, workspace }: PageHeaderProps) {
         inputHandleRef={titleInputHandleRef}
         pageId={page?.id}
         docCollection={workspace.docCollection}
+        onTriggerEdit={onTriggerEdit}
```
Comment (forehalo) [inline packages/frontend/core/src/pages/workspace/detail-page/detail-page-header.tsx]: this should be triggered when saving edits

File: packages/frontend/core/src/mixpanel/events.ts [hunk 11]
```diff
@@ -35,7 +35,13 @@ type DocEvents =
   | 'renameDoc'
   | 'linkDoc'
   | 'deleteDoc'
-  | 'switchPageMode';
+  | 'switchPageMode'
+  | 'openDocOptionsMenu'
+  | 'openDocInfo'
+  | 'editDocTitle'
```
Comment (forehalo) [inline packages/frontend/core/src/mixpanel/events.ts]: renameDoc

File: packages/frontend/core/src/mixpanel/events.ts [hunk 12]
```diff
@@ -35,7 +35,13 @@ type DocEvents =
...
+  | 'moveToTrash'
+  | 'viewHistoryVersions'
```
Comment (forehalo) [inline packages/frontend/core/src/mixpanel/events.ts:43]: `$.header.history.open` `$.header.pageInfo.open`

File: packages/frontend/core/src/components/affine/share-page-modal/index.tsx [hunk 13]
```diff
@@ -11,6 +13,11 @@ type SharePageModalProps = {
 
 export const SharePageButton = ({ workspace, page }: SharePageModalProps) => {
   const confirmEnableCloud = useEnableCloud();
+  const handleOpenShareModal = useCallback((open: boolean) => {
+    if (open) {
+      track.$.header.actions.openShareMenu();
```
Comment (forehalo) [inline packages/frontend/core/src/components/affine/share-page-modal/index.tsx]: `$.sharePanel.$.open()`

File: packages/frontend/core/src/components/blocksuite/block-suite-header/info/index.tsx [hunk 14]
```diff
@@ -1,15 +1,20 @@
...
 import { useSetAtom } from 'jotai';
+import { useCallback } from 'react';
 
 export const InfoButton = () => {
   const setOpenInfoModal = useSetAtom(openInfoModalAtom);
   const t = useI18n();
-  const onOpenInfoModal = () => {
+
+  const onOpenInfoModal = useCallback(() => {
+    track.$.header.actions.openDocInfo();
```
Comment (forehalo) [inline packages/frontend/core/src/components/blocksuite/block-suite-header/info/index.tsx:14]: `$.docInfoPanel.$.open`

File: packages/frontend/core/src/components/blocksuite/block-suite-header/favorite/index.tsx (hunk 7 above)
Comment (JimmFly) [reply packages/frontend/core/src/components/blocksuite/block-suite-header/favorite/index.tsx:35]: I tried this, but it didn't work.
`data-event-props="$.header.actions.toggleFavorite"`

Comment (JimmFly) [reply packages/frontend/core/src/components/blocksuite/block-suite-header/favorite/index.tsx:35]: It looks like a problem with IconButton

**Review by forehalo — APPROVED** [review]

**Pull request merged**